### PR TITLE
Add release notes section to UPGRADING document.

### LINKED ISSUE #865: Create release note documenting upgrades with haas -> hil name change
For the 0.2 release, we need to document that:

- [ ] Since the systemd script is created manually, the old one needs to manually be deleted and the new one copied in.
- [ ]  If wanting to keep the previous `haas` user, then the script needs to be modified to reflect the different username.
- [ ] The `/var/lib/haas` and `/etc/haas.cfg` entries need to be moved (or at least symlinked)
- [ ] You should remove the "haas" version, since it will be a different set of scripts: `pip uninstall haas`
- [ ] Re-copy `hil.wsgi` and update apache's wsgi.conf entry to point to it.
- [ ] Update any scripts that have env vars (like `HAAS_ENDPOINT`) to their HIL_ varieties.

### PR DESCRIPTION
**Pull request by zenhack**
Fixes #865 

Note that the 0.3 section is not necessarily complete; I added stuff I could think of off the top of my head, and skimmed the closed issues on the 0.3 milestone (where I didn't find anything that seemed like it needed mentioning).

### CODE REVIEW

File: docs/UPGRADING.rst [hunk 1]
```diff
@@ -34,3 +35,38 @@ This file describes the procedure for upgrading HIL to a new version.
 
      $ systemctl start httpd
      $ systemctl start hil_network
+
+Release Notes
+-------------
+
+0.3 (Upcoming)
```
Comment (xuhang57) [inline docs/UPGRADING.rst]: Upcoming is a bit vague to me. Maybe having an actual date?

File: docs/UPGRADING.rst [hunk 2]
```diff
@@ -34,3 +35,38 @@ This file describes the procedure for upgrading HIL to a new version.
...
+++++++++++++++
+
+* Validation checks for the configuration file have been added; be advised
+  that if there were errors in your ``hil.cfg`` that were previously
+  undetected you may need to fix them before the servers will start again
+* The interface to the ``hil`` command line tool has changed substantially;
```
Comment (xuhang57) [inline docs/UPGRADING.rst]: maybe we need to have an updated doc so we can point the readers to the relevant documentation?

File: docs/UPGRADING.rst (hunk 1 above)
Comment (naved001) [reply docs/UPGRADING.rst]: we don't really have an actual date. it's just as soon as we are done with all our issues marked with 0.3 milestone.

**Comment by naved001** [comment]
Some more things that might be useful to put in the changelog. 
* All HIL APIs now have client library support. 
* Support for maintenance pool.
* Some new APIs
  - Networking actions can now be queried to get their status. 
  - List users when using database backend
* Updated APIs
  - List networks now shows public networks to regular users.
* Some switches now support public key authentication. See `docs/network-drivers.md` for details.

File: docs/UPGRADING.rst (hunk 1 above)
Comment (xuhang57) [reply docs/UPGRADING.rst]: sure, am working on the version PR, which is the last PR for 0.3 milestones I believe

**Comment by naved001** [comment]
I am happy.

File: docs/UPGRADING.rst (hunk 2 above)
Comment (naved001) [reply docs/UPGRADING.rst]: I think just mentioning `--help` should be enough. We could probably, in the future, add some documentation about the CLI (even a man page for HIL `man hil`).

File: docs/UPGRADING.rst [hunk 3]
```diff
@@ -34,3 +35,51 @@ This file describes the procedure for upgrading HIL to a new version.
...
+  If you have custom scripts that invoke it they will likely need to be
+  modified.
+* URLs for API calls are now prefixed with a version number, which for the
+  current (unstable) API is ``v0``; you may need to update scripts and
+  settings accordingly (although the HIL tools themselves should remaining
```
Comment (naved001) [inline docs/UPGRADING.rst]: typo: should **remain** internally consistent

File: docs/UPGRADING.rst [hunk 4]
```diff
@@ -34,3 +35,51 @@ This file describes the procedure for upgrading HIL to a new version.
...
+* Updated APIs:
+  * List networks now shows public networks to regular users.
+  * Some switches now support public key authentication. See
+    ``docs/network-drivers.md`` for details.
+
+0.2
++++
+
+HaaS was renamed to HIL in this release. Accordingly, there are several
+changess that need to be made when upgrading:
```
Comment (naved001) [inline docs/UPGRADING.rst]: typo: changess -> changes

**Comment by zenhack** [comment]
@naved, fixed.

**Comment by naved001** [comment]
not sure if we should merge it now, or after the last PR for 0.3 is merged?

**Comment by zenhack** [comment]
I don't see any reason not to do so now.

Quoting Naved Ansari (2018-04-03 17:47:13)

>    not sure if we should merge it now, or after the last PR for 0.3 is
>    merged?

**Pull request merged**